quic: flaky reuse tests

```
=== RUN   TestReuseListenOnAllIPv4
      reuse_test.go:59: 
          	Error Trace:	reuse_test.go:59
          	Error:      	Condition never satisfied
          	Test:       	TestReuseListenOnAllIPv4
          	Messages:   	expected garbage collector to be running
  --- FAIL: TestReuseListenOnAllIPv4 (0.14s)

--- a/p2p/transport/quic/reuse_test.go
+++ b/p2p/transport/quic/reuse_test.go
@@ -56,7 +56,7 @@ func cleanup(t *testing.T, reuse *reuse) {
 
 func TestReuseListenOnAllIPv4(t *testing.T) {
 	reuse := newReuse()
-	require.Eventually(t, isGarbageCollectorRunning, 100*time.Millisecond, time.Millisecond, "expected garbage collector to be running")
+	require.Eventually(t, isGarbageCollectorRunning, 500*time.Millisecond, 50*time.Millisecond, "expected garbage collector to be running")
 	cleanup(t, reuse)
 
 	addr, err := net.ResolveUDPAddr("udp4", "0.0.0.0:0")
@@ -68,7 +68,7 @@ func TestReuseListenOnAllIPv4(t *testing.T) {
 
 func TestReuseListenOnAllIPv6(t *testing.T) {
 	reuse := newReuse()
-	require.Eventually(t, isGarbageCollectorRunning, 100*time.Millisecond, time.Millisecond, "expected garbage collector to be running")
+	require.Eventually(t, isGarbageCollectorRunning, 500*time.Millisecond, 50*time.Millisecond, "expected garbage collector to be running")
 	cleanup(t, reuse)
 
 	addr, err := net.ResolveUDPAddr("udp6", "[::]:1234")

